### ISSUE TITLE
UI on the permissions page

### ISSUE THREAD

**Issue by Abby-Wheelis** [e-mission/e-mission-docs#919]
I've had the chance to help a few users through the initial hurdle of getting permissions enabled, and noted a few themes that we should keep in mind when we next update that page:

1. lack of awareness or understanding that the fixes need to be initiated by the app - as a feature of ensuring consent
2. confusion around the "categories" of permissions - ie there's two steps to "fixing" location and having that banner go green
3. the fix/refresh buttons did not use to look like buttons

1 and 2 could potentially be addressed by re-wording or adding to the statements to the user
3 was addressed already here: [5b3dc298ff0194983691af40e585db7e1156b9fb](https://github.com/e-mission/e-mission-phone/commit/5b3dc298ff0194983691af40e585db7e1156b9fb)

All are good to keep in mind as we soon update the page and potentially find ways to configure the UI in a way that "fixing" the issues and clearing the page is as intuitive and pain-free as possible, letting people access the app!

One idea I'd had was a popup of some sort that could be presented upon entering the page and then accessed again from some button, acting as a "tutorial" or "instructions" page guiding the user. The use of something that is shown and hidden would allow for detailed instructions to be referenced as needed without cluttering the page itself.

**Comment by Abby-Wheelis** [comment]
As an update to this issue, I've created a new `AppStatusModal` in my ongoing migration of the Profile screen, which incorporates some of this feedback. One thing I've chosen to do is to create a single list of the permissions the device requires, and a separate screen for the explanations, this at least declutters the screen to an extent. See screenshots below, note that both modals are a fixed size but are scrollable:
<img src="https://github.com/e-mission/e-mission-docs/assets/54848919/993cf33e-0b17-4491-b683-35668a777c92" width="300"  />                         <img src="https://github.com/e-mission/e-mission-docs/assets/54848919/1f5f980d-6597-43ed-9082-7bf092c5ddf5" width="300"  />
An ongoing consideration is how we can limit the work a user has to do in order to get the app up and running, and thereby increase the number of people who want to or are willing to use the app.

**Comment by shankari** [comment]
> One idea I'd had was a popup of some sort that could be presented upon entering the page and then accessed again from some button, acting as a "tutorial" or "instructions" page guiding the user. The use of something that is shown and hidden would allow for detailed instructions to be referenced as needed without cluttering the page itself.

Were we going to consider this - some kind of "wizard" view? Would having one permission on one "page" seems to be pretty standard for new phone onboarding, at least. But I'm not sure if it will make things too complicated for the number of permissions that we need

**Comment by Abby-Wheelis** [comment]
With a tutorial type view here, I think if we have to show users how to use it, then the page needs to be redesigned more. I think all we'd need to convey is "if it's red click the fix button and follow the instructions", right? I'm hoping that getting rid of the multi-dropdown configuration and isolating the explanations to their own Modal (with a clear way to access it at the top, maintaining transparency) will make the screen simple enough to lead users to the conclusion of "if it's red click fix" on their own. 

I also wonder if the actual device settings pages are something that's tripping up users just as much, at least in some cases it takes you to a page in the device settings, and you kind of have to know what to do from there. Examples off the top of my head would be in iOS you get dropped in the app permissions, but then you have to click locations, then, always, and toggle on precise. Android to turn off optimizations sometimes I've had to change a dropdown to not optimized, then scroll a while... I can see where this could get confusing and/or burdensome ... if we could "wizard" there and show users how to fix things in their device's settings that would be pretty magical, but this would differ per device and OS and I'm not sure iOS or Android would let us control the screen outside of our own app. 

Best case scenario I think would be that every permission "fix" would just be a popup where they have to click "allow", but not sure to what extent that's in our control.

**Comment by shankari** [comment]
> Best case scenario I think would be that every permission "fix" would just be a popup where they have to click "allow", but not sure to what extent that's in our control.

Yeah, woudn't that be great! That is what it used to be back in the old days. In fact in the super old days, on android, the permissions were listed in the app store page, and you didn't have *any* dynamic popups. But the platforms have slowly been cracking down on this - they _want to_ add more friction so that the user has to really really want this functionality to turn it on.

See discussion around the optimization screen in particular in the API upgrade issue (https://github.com/e-mission/e-mission-docs/issues/934#issuecomment-1681622555)

Having said that, if you can come up with alternatives (e.g. `ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS`), I am happy to try them, although not on time-sensitive releases, in case the update is rejected.

**Comment by JGreenlee** [comment]
> Best case scenario I think would be that every permission "fix" would just be a popup where they have to click "allow", but not sure to what extent that's in our control.

I do think this is achievable. Other apps manage to get the permissions they need without making users dig around in any system settings pages -- even apps that have similarly extensive background services and location permission requirements.
Life360 is a good example - when I get a chance sometime, I'll screenrecord the permissions flow on my personal phone

**Comment by Abby-Wheelis** [comment]
The onboarding changes have now been merged into production, and updates like [battery optimizations](https://github.com/e-mission/e-mission-data-collection/pull/209) and [location permissions](https://github.com/e-mission/e-mission-data-collection/pull/210https://github.com/e-mission/e-mission-data-collection/pull/210) are decreasing the barriers to install by adding popups where users used to have to navigate through complex settings.
